zeppelin-interpreter/src/main/java/org/apache/zeppelin/display/Input.java

```diff
@@ -86,6 +86,34 @@ public String getArgument() {
     return argument;
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    Input<?> input = (Input<?>) o;
+
+    if (hidden != input.hidden) return false;
+    if (name != null ? !name.equals(input.name) : input.name != null) return false;
+    if (displayName != null ? !displayName.equals(input.displayName) : input.displayName != null)
+      return false;
+    if (defaultValue != null ?
+        !defaultValue.equals(input.defaultValue) : input.defaultValue != null)
+      return false;
```

Comment:
nit: this is a bit hard to read. I don't know what style we use here, but consider
```
if () {
}
```